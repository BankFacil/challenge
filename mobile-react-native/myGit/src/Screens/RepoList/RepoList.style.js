import { Dimensions, Animated } from 'react-native';
import styled from 'styled-components/native';

import { Layout, Margins } from '../../Styleguide';

const { width, height } = Dimensions.get('window');

export const Wrapper = styled.View`
  ${Layout.Global.wrapper};
  padding: 8px;
`;

export const Logo = styled(Animated.Image)`
  height: 60px;
  align-self: center;
  resizeMode: contain;
`;

