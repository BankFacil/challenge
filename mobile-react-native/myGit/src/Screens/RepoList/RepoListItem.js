import React from 'react';
import PropTypes from 'prop-types';
import {
  Image,
  TouchableWithoutFeedback,
  View,
  Text,
} from 'react-native';
import * as Animatable from 'react-native-animatable';
// import { withNavigationFocus } from 'react-navigation';
// import { Transition } from 'react-navigation-fluid-transitions';

// import { Text } from 'components/styleguide';

{/* <Transition shared={`gym-container-${slug}`}>
<View style={styles.container} />
</Transition> */}

const RepoListItem = ({
  index,
  repo: { name, description },
  onPress,
}) =>
  (
    <TouchableWithoutFeedback
      onPress={onPress}
      testID={`RepoItem-${index}`}
    >
      <Animatable.View
        animation="fadeInUp"
        // duration={animationDuration}
        delay={index * 200}
        useNativeDriver
        style={{ width: '100%', borderBottomWidth: 1 }}
      >
        <Text style={{ fontSize: 21, color: '#212121' }}>{name}</Text>
        <View style={{ borderWidth: 1, alignItems: 'center', padding: 24, marginTop: 4, marginBottom: 24 }}>
          <Text>{description}</Text>
        </View>
      </Animatable.View>
    </TouchableWithoutFeedback>
  );

  RepoListItem.propTypes = {
    index: PropTypes.number.isRequired,
    onPress: PropTypes.func.isRequired,
    repo: PropTypes.shape({
      name: PropTypes.string,
      description: PropTypes.string,
    }).isRequired,
};

export default RepoListItem;
