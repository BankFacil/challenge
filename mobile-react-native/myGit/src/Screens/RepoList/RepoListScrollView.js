import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  ScrollView,
  Vibration,
  View,
  ActivityIndicator
} from 'react-native';
import * as Animatable from 'react-native-animatable';

import { configureLayoutAnimation } from '../../Utils/layoutAnimation';

import RepoListItem from './RepoListItem';

class RepoListScrollView extends React.Component {
  static propTypes = {
    repos: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string,
      description: PropTypes.string,
    })).isRequired,
    getRepos: PropTypes.func.isRequired,
  };

  static defaultProps = {
    repos: [],
  };

  state = {
    isRefreshing: false,
  };

  onScrollEndDrag = (event) => {
    const { contentOffset } = event.nativeEvent;

    if (contentOffset.y < -40 && !this.state.isRefreshing) {
      Vibration.vibrate();
      this.setState({ isRefreshing: true },
        () => setTimeout(() => {
          this.props.getRepos().then(() => {
            this.setState({ isRefreshing: false });
            configureLayoutAnimation();
          });
        }, 1500),
      );
    }
  }

  render() {
    const { isRefreshing } = this.state;
    const { repos, onScroll } = this.props;

    return (
      <ScrollView
        scrollEventThrottle={100}
        onScrollEndDrag={this.onScrollEndDrag}
        scrollEventThrottle={16}
        onScroll={onScroll}
      >
        {isRefreshing && (
          <Animatable.View animation="fadeIn">
            <ActivityIndicator />
          </Animatable.View>
        )}
        {repos.map((repo, index) => (
          <RepoListItem
            key={repo.name}
            repo={repo}
            index={index}
            onPress={() => {}}
          />
        ))}
      </ScrollView>
    );
  }
}

export default RepoListScrollView;
