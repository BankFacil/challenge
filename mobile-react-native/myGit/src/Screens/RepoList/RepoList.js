import React from 'react';
import PropTypes from 'prop-types';
import {
  ActivityIndicator,
  AsyncStorage,
  StatusBar,
  StyleSheet,
  View,
  TextInput,
} from 'react-native';
import { connect } from 'react-redux';

import RepoListScrollView from './RepoListScrollView';

import { getRepos } from '../../Store/Ducks/repository';

import {
  Wrapper,
  RepoLogo,
} from './RepoList.style';

export class RepoList extends React.Component {
  static propTypes = {
    repos: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    })),
    getRepos: PropTypes.func.isRequired,
    isLoading: PropTypes.bool
  }

  static defaultProps = {
    isLoading: false,
    repos: [],
    getRepos: () => {}
  }

  state = {
    keyword: ''
  }

  componentDidMount() {
    this.props.getRepos();
  }

  onSearch(keyword) {
    this.setState({ keyword });
  }

  getFilteredRepos(repos) {
    const { keyword } = this.state;
    if (repos.length) {
      return repos.filter(repo => (repo.name.toLowerCase()).includes(keyword.toLowerCase()))
    }
    return repos;
  }

  render() {
    const { repos, isLoading } = this.props;

    return (
      <Wrapper>
        <StatusBar barStyle="default" />
        <RepoLogo source={require('../../Assets/github-logo-full.png')} />
        <TextInput
          onChange={event => this.onSearch(event.nativeEvent.text)}
          placeholder="Repository Search"
          style={{ borderWidth: 1, borderRadius: 4, width: '100%', padding: 8, marginBottom: 40 }} />
        <RepoListScrollView
          isLoading={isLoading}
          getRepos={this.props.getRepos}
          repos={this.getFilteredRepos(repos)}
        />
      </Wrapper>
    );
  }
}

const mapStateToProps = state => ({
  repos: state.repository.repos || [],
  isLoading: state.repository.loading,
});

const mapDispatchToProps = {
  getRepos,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(RepoList);
