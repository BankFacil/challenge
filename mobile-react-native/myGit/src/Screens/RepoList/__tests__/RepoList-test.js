/**
 * @format
 */

import 'react-native';
import React from 'react';
import { RepoList } from '../RepoList';
import renderer from 'react-test-renderer';

it('renders RepoList correctly (snapshot)', () => {
  const component = renderer.create(<RepoList />).toJSON();
  expect(component).toMatchSnapshot();
});
