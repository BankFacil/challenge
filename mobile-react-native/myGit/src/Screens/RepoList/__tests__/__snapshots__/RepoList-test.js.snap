// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RepoList correctly (snapshot) 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginTop": 20,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/Assets/github-logo-full.png",
      }
    }
    style={
      Array [
        Object {
          "alignSelf": "center",
          "height": 150,
          "marginBottom": 10,
          "resizeMode": "contain",
          "width": 375,
        },
      ]
    }
  />
  <TextInput
    allowFontScaling={true}
    onChange={[Function]}
    placeholder="Repository Search"
    rejectResponderTermination={true}
    style={
      Object {
        "borderRadius": 4,
        "borderWidth": 1,
        "marginBottom": 40,
        "padding": 8,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <RCTScrollView
    onScrollEndDrag={[Function]}
    scrollEventThrottle={100}
  >
    <View />
  </RCTScrollView>
</View>
`;
